### ISSUE TITLE
after node update thumbs ups fails and cannot be reinstalled

### ISSUE THREAD

**Issue by prearisto** [thumbsup/thumbsup#166]
## Bug report

Install as an npm package:

```
Thumbsup version: latest
Node version: 12.1.0
NPM version: 6.9.0
Operating system: Mojave
```

About the bug...
Installing fails, works only when downgrade node to version 10, and there it shows a warning of deprecation, which resulted in error in later version:
npm install -g thumbsup
/usr/local/bin/thumbsup -> /usr/local/lib/node_modules/thumbsup/bin/thumbsup.js

> integer@1.0.7 install /usr/local/lib/node_modules/thumbsup/node_modules/integer
> node-gyp rebuild

  CXX(target) Release/obj.target/integer/src/integer.o
../src/integer.cpp:329:21: error: no matching constructor for initialization of 'v8::String::Value'
                v8::String::Value utf16(string);
                                  ^     ~~~~~~
/Users/elinemtsov/.node-gyp/12.1.0/include/node/v8.h:3025:5: note: candidate constructor not viable: no known conversion from 'v8::Local<v8::String>' to
      'const v8::String::Value' for 1st argument
    Value(const Value&) = delete;
    ^
/Users/elinemtsov/.node-gyp/12.1.0/include/node/v8.h:3018:5: note: candidate constructor not viable: requires 2 arguments, but 1 was provided
    Value(Isolate* isolate, Local<v8::Value> obj);
    ^
1 error generated.
make: *** [Release/obj.target/integer/src/integer.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:196:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:256:12)
gyp ERR! System Darwin 18.5.0
gyp ERR! command "/usr/local/Cellar/node/12.1.0/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /usr/local/lib/node_modules/thumbsup/node_modules/integer
gyp ERR! node -v v12.1.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! integer@1.0.7 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the integer@1.0.7 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

With Node@10 there is a warning:
npm install -g thumbsup
/usr/local/bin/thumbsup -> /usr/local/lib/node_modules/thumbsup/bin/thumbsup.js

> integer@1.0.7 install /usr/local/lib/node_modules/thumbsup/node_modules/integer
> node-gyp rebuild

  CXX(target) Release/obj.target/integer/src/integer.o
../src/integer.cpp:329:21: warning: 'Value' is deprecated [-Wdeprecated-declarations]
                v8::String::Value utf16(string);
                                  ^
/Users/elinemtsov/.node-gyp/10.15.3/include/node/v8.h:2916:5: note: 'Value' has been explicitly marked deprecated here
    V8_DEPRECATED("Use Isolate version", explicit Value(Local<v8::Value> obj));
    ^
/Users/elinemtsov/.node-gyp/10.15.3/include/node/v8config.h:327:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
1 warning generated.
  SOLINK_MODULE(target) Release/integer.node

> better-sqlite3@4.2.0 install /usr/local/lib/node_modules/thumbsup/node_modules/better-sqlite3
> node-gyp rebuild

  ACTION deps_sqlite3_gyp_action_before_build_target_unpack_sqlite_dep Release/obj/gen/sqlite-autoconf-3240000/sqlite3.c
  TOUCH Release/obj.target/deps/action_before_build.stamp
  CC(target) Release/obj.target/sqlite3/gen/sqlite-autoconf-3240000/sqlite3.o
  LIBTOOL-STATIC Release/sqlite3.a
  CXX(target) Release/obj.target/better_sqlite3/src/better_sqlite3.o
./src/util/database-handles.lzz:14:39: warning: 'Utf8Value' is deprecated [-Wdeprecated-declarations]
                v8::String::Utf8Value utf8(filename);
                                      ^
/Users/elinemtsov/.node-gyp/10.15.3/include/node/v8.h:2891:5: note: 'Utf8Value' has been explicitly marked deprecated here
    V8_DEPRECATED("Use Isolate version",
    ^
/Users/elinemtsov/.node-gyp/10.15.3/include/node/v8config.h:327:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
./src/util/sql-string.lzz:44:5: warning: 'Value' is deprecated [-Wdeprecated-declarations]
  : value (string), missing_semicolon (value.length() < 1 || (*value)[value.length() - 1] != 0x3b)
    ^
/Users/elinemtsov/.node-gyp/10.15.3/include/node/v8.h:2916:5: note: 'Value' has been explicitly marked deprecated here
    V8_DEPRECATED("Use Isolate version", explicit Value(Local<v8::Value> obj));
    ^
/Users/elinemtsov/.node-gyp/10.15.3/include/node/v8config.h:327:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
./src/objects/database.lzz:146:39: warning: 'Utf8Value' is deprecated [-Wdeprecated-declarations]
                v8::String::Utf8Value sql(source);
                                      ^
/Users/elinemtsov/.node-gyp/10.15.3/include/node/v8.h:2891:5: note: 'Utf8Value' has been explicitly marked deprecated here
    V8_DEPRECATED("Use Isolate version",
    ^
/Users/elinemtsov/.node-gyp/10.15.3/include/node/v8config.h:327:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
./src/objects/database.lzz:165:47: warning: 'Utf8Value' is deprecated [-Wdeprecated-declarations]
                        v8::String::Utf8Value only_database(onlyDatabase);
                                              ^
/Users/elinemtsov/.node-gyp/10.15.3/include/node/v8.h:2891:5: note: 'Utf8Value' has been explicitly marked deprecated here
    V8_DEPRECATED("Use Isolate version",
    ^
/Users/elinemtsov/.node-gyp/10.15.3/include/node/v8config.h:327:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
./src/objects/database.lzz:211:39: warning: 'Utf8Value' is deprecated [-Wdeprecated-declarations]
                v8::String::Utf8Value name(nameString);
                                      ^
/Users/elinemtsov/.node-gyp/10.15.3/include/node/v8.h:2891:5: note: 'Utf8Value' has been explicitly marked deprecated here
    V8_DEPRECATED("Use Isolate version",
    ^
/Users/elinemtsov/.node-gyp/10.15.3/include/node/v8config.h:327:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
./src/objects/database.lzz:230:39: warning: 'Utf8Value' is deprecated [-Wdeprecated-declarations]
                v8::String::Utf8Value filename(filenameString);
                                      ^
/Users/elinemtsov/.node-gyp/10.15.3/include/node/v8.h:2891:5: note: 'Utf8Value' has been explicitly marked deprecated here
    V8_DEPRECATED("Use Isolate version",
    ^
/Users/elinemtsov/.node-gyp/10.15.3/include/node/v8config.h:327:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
./src/objects/statement.lzz:89:35: warning: 'Value' is deprecated [-Wdeprecated-declarations]
                v8::String::Value sql(source);
                                  ^
/Users/elinemtsov/.node-gyp/10.15.3/include/node/v8.h:2916:5: note: 'Value' has been explicitly marked deprecated here
    V8_DEPRECATED("Use Isolate version", explicit Value(Local<v8::Value> obj));
    ^
/Users/elinemtsov/.node-gyp/10.15.3/include/node/v8config.h:327:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
./src/util/data.lzz:86:222: warning: 'Utf8Value' is deprecated [-Wdeprecated-declarations]
  ...< v8 :: Number > :: Cast ( value ) -> Value ( ) ) ; } else if ( value -> IsString ( ) ) { v8 :: String :: Utf8Value utf8 ( v8 :: Local < v8 :: String > :: Cast ( valu...
                                                                                                                         ^
/Users/elinemtsov/.node-gyp/10.15.3/include/node/v8.h:2891:5: note: 'Utf8Value' has been explicitly marked deprecated here
    V8_DEPRECATED("Use Isolate version",
    ^
/Users/elinemtsov/.node-gyp/10.15.3/include/node/v8config.h:327:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
./src/util/data.lzz:91:213: warning: 'Utf8Value' is deprecated [-Wdeprecated-declarations]
  ...< v8 :: Number > :: Cast ( value ) -> Value ( ) ) ; } else if ( value -> IsString ( ) ) { v8 :: String :: Utf8Value utf8 ( v8 :: Local < v8 :: String > :: Cast ( valu...
                                                                                                                         ^
/Users/elinemtsov/.node-gyp/10.15.3/include/node/v8.h:2891:5: note: 'Utf8Value' has been explicitly marked deprecated here
    V8_DEPRECATED("Use Isolate version",
    ^
/Users/elinemtsov/.node-gyp/10.15.3/include/node/v8config.h:327:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
./src/binding/binder.lzz:119:55: warning: 'Utf8Value' is deprecated [-Wdeprecated-declarations]
                                v8::String::Utf8Value param_name(key);
                                                      ^
/Users/elinemtsov/.node-gyp/10.15.3/include/node/v8.h:2891:5: note: 'Utf8Value' has been explicitly marked deprecated here
    V8_DEPRECATED("Use Isolate version",
    ^
/Users/elinemtsov/.node-gyp/10.15.3/include/node/v8config.h:327:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
./src/binding/multi-binder.lzz:76:55: warning: 'Utf8Value' is deprecated [-Wdeprecated-declarations]
                                v8::String::Utf8Value param_name(key);
                                                      ^
/Users/elinemtsov/.node-gyp/10.15.3/include/node/v8.h:2891:5: note: 'Utf8Value' has been explicitly marked deprecated here
    V8_DEPRECATED("Use Isolate version",
    ^
/Users/elinemtsov/.node-gyp/10.15.3/include/node/v8config.h:327:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
11 warnings generated.
  SOLINK_MODULE(target) Release/better_sqlite3.node
  CC(target) Release/obj.target/test_extension/deps/test_extension.o
  SOLINK_MODULE(target) Release/test_extension.node
+ thumbsup@2.10.0
added 395 packages from 247 contributors in 81.644s

**Comment by emilyst** [comment]
`thumbsup` appears to depend on `integer@1.0.7` which doesn't build on macOS Mojave 10.14.5. The latest version of `integer` does build fine. It's likely this can be fixed by upgrading that dependency.

**Comment by rprieto** [comment]
Hi @prearisto and @emilyst, thanks for all the troubleshooting! I just published `v2.10.1` which uses the latest version of `better-sqlite`, can you let me know if it works now?

The only downside is that it drops support for Node 4, but it's out of LTS support now so it seems acceptable.

**Comment by emilyst** [comment]
Installs fine. Thanks.
